lms/djangoapps/email_marketing/tasks.py

```diff
@@ -294,6 +299,71 @@ def _get_course_content(course_url, sailthru_client, email_config):
     return response
 
 
+def _get_or_create_sailthru_list_for_site(sailthru_client, email_config, site=None, default_list_name=None):
+    """
+    Get the user list name from cache if exists else create one and return the name,
+    callers of this function should perform the enabled check of email config.
+    :param sailthru_client:
+    :param email_config:
+    :param site:
+    :param default_list_name
+    :return: list name if exists or created else return None
+    """
+    if site and site.id != settings.SITE_ID:
+        list_name = site.domain.replace(".", "_") + "_user_list"
+    else:
+        list_name = default_list_name
+
+    _get_or_create_sailthru_list(sailthru_client, email_config, list_name)
+    return list_name
+
+
+def _get_or_create_sailthru_list(sailthru_client, email_config, list_name):
+    """
+    Get list from sailthru and return if list_name exists else create a new one
+    and return list data for all lists.
+    """
+    cached = cache.get(email_config.CACHE_KEY)
```

Comment:
I would rename `CACHE_KEY` to `SAILTHRU_LIST_CACHE_KEY` and move it from `EmailMarketingConfiguration ` to this module. If you do that you will not have to pass the email_config as a parameter to the previous two functions in the stack.